python/cuml/experimental/explainer/tree_shap.pyx

```diff
@@ -100,21 +111,27 @@ class TreeExplainer:
     def __init__(self, *, model):
         # Handle various kinds of tree model objects
         cls = model.__class__
+        cls_module, cls_name = cls.__module__, cls.__name__
         # XGBoost model object
-        if cls.__module__ == 'xgboost.core' and cls.__name__ == 'Booster':
+        if re.match(r'xgboost.*$', cls_module) and cls_name == 'Booster':
```

Comment:
> this type of code for guessing module names ends up being fragile over time. Not sure how to avoid it though.

The alternative is to ask the user to explicitly indicate which model type it is. This will be more typing however.